### PR TITLE
replace max_msg_size with max_transfer_size

### PR DESCRIPTION
**Pull request by jlevon**

### CODE REVIEW

File: docs/vfio-user.rst [hunk 1]
```diff
@@ -317,9 +317,8 @@ Message sizes
 
 Some requests have an ``argsz`` field. In a request, it defines the maximum
 expected reply payload size, which should be at least the size of the fixed
-reply payload headers defined here. For messages that don't have a reply
-payload, it defines the size of the incoming request (not including the standard
-header); otherwise it's not related to the request message size.
+reply payload headers defined here. The *request* payload size is defined by the
+usual ``msg_size`` field in the header, not this field.
```
[the remark this answers is not in the document]
Comment (jlevon) [reply docs/vfio-user.rst:321]: aye

File: docs/vfio-user.rst [hunk 2]
```diff
@@ -328,6 +327,11 @@ included in the reply, but the ``argsz`` field in the reply indicates the needed
 size, allowing a client to allocate a larger buffer for holding the reply before
 trying again.
 
+In addition, during negotiation (see  `Version`_), the client and server may
+each specify a ``max_transfer_size`` value; this defines the maximum data that
```
[the remark this answers is not in the document]
Comment (jlevon) [reply docs/vfio-user.rst:331]: I'm not sure I'm convinced: I don't really think of "transferring" the message as a whole. max_data_transfer_size is a bit long?

File: lib/libvfio-user.c [hunk 3]
```diff
@@ -1606,79 +1616,139 @@ vfu_unmap_sg(vfu_ctx_t *vfu_ctx, const dma_sg_t *sg, struct iovec *iov, int cnt)
...
-        if (errno == ENOMSG) {
-            vfu_reset_ctx(vfu_ctx, "closed");
-            errno = ENOTCONN;
-        } else if (errno == ECONNRESET) {
-            vfu_reset_ctx(vfu_ctx, "reset");
-            errno = ENOTCONN;
+    remaining = sg->length;
+    count = 0;
+
+    while (remaining > 0) {
```
[the remark this answers is not in the document]
Comment (jlevon) [reply lib/libvfio-user.c:1709]: let me take a stab at it

File: docs/vfio-user.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (tmakatos) [reply docs/vfio-user.rst:331]: Independently of whether or not `data` is included, `transfer` could become `xfer`?

[the remark this answers is not in the document]
Comment (jlevon) [reply docs/vfio-user.rst]: ok, then, I did both.

[the remark this answers is not in the document]
Comment (swapnili) [reply docs/vfio-user.rst]: Ok max_transfer_size is also fine by me

**Review by swapnili — APPROVED** [review]

**Pull request merged**